### ISSUE TITLE
Rule Proposal: `no-empty-function`

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#5161]
```js
const getEmptyObject = () => {};
console.log(getEmptyObject()); // undefined
```

If the body of an arrow function is `{}`, it's a block statement, not an empty object literal.
This might confuse developers.

This proposal is that we add the option to warn such empty arrow function to the `no-empty` rule.

----

The following patterns are considered problems:

```js
/* eslint no-empty: [2, {arrowFunctions: true}] */

const f = () => {};  /*error Empty block statement.*/
```

The following patterns are not considered problems:

```js
/* eslint no-empty: [2, {arrowFunctions: true}] */

// this is an empty object literal.
const f = () => ({});

// there is an clear comment.
const f = () => {
    // do nohting.
};
```

**Comment by nzakas** [comment]
Sounds good

**Comment by mysticatea** [comment]
From https://github.com/eslint/eslint/pull/5174#issuecomment-181991929, I updated this proposal.

https://gist.github.com/mysticatea/4e0141ce9148186e93a8

The new proposal has `"allow"` option to substitute for the `"methods"` option of `no-empty` rule and the behavior of old `arrow-body-style` rule.

**Comment by nzakas** [comment]
I thought we were going to do a `no-empty-function` rule instead of modifying `no-empty`?

**Comment by mysticatea** [comment]
Yes.

**Comment by mysticatea** [comment]
Yes.

I will add new `no-empty-function` rule and remove the `methods` option from `no-empty` rule.
I don't add `arrowFunctions` option to `no-empty` rule.
